service/index.js

```diff
@@ -240,6 +265,40 @@ module.exports = class AuthenticationService {
       if (!deletedCount && (strict === true)) return Promise.reject(new Error(`Failed to remove user: ${email}.`));
     });
   }
+
+  /**
+   * Authorizes third party then logs in or signs up the user.
+   */
+  authorizeThirdParty(code, provider) {
+    if (!code || !provider) {
+      return Promise.reject(
+        new Error('Required parameters missing for third party authorization.')
+      );
+    }
+
+    return p(this).deferrari.deferUntil(THIRD_PARTY_CONFIG)
+    .then(authieThirdParty => authieThirdParty.login(code, provider)
+      .then(result => p(this).deferrari.deferUntil(CONNECTED)
+        .then(models => models.AuthUser.findOne({where: {email: result.email}})
```

Comment:
What if they use a separate email for `facebook` and `google`, or a separate one in either from `authie`? We don't want 3 separate `AuthUsers`. We should have an optional param that a client can pass with a third party token being the `email` or `authUser.id`, that way if we're allowing them to link an account from an already authenticated authUser, we can force it to not create a new authUser based off of a different email.